libdleyna/core/connector-mgr.c

```diff
@@ -34,33 +34,54 @@ const dleyna_connector_t *dleyna_connector_mgr_load(const gchar *name)
 	const dleyna_connector_t *connector;
 	dleyna_connector_get_interface_t get_interface;
 	gchar *path;
+	const gchar *connector_path;
+	gchar **connector_path_list;
+	gsize i;
 
 	DLEYNA_LOG_DEBUG("Enter");
 
-	path = g_strdup_printf("%s/%s%s.so", CONNECTOR_DIR,
-			       DLEYNA_CONNECTOR_LIB_PATTERN, name);
-	module = g_module_open(path, G_MODULE_BIND_LAZY);
-	g_free(path);
+	connector_path = g_getenv ("DLEYNA_CONNECTOR_PATH");
+	if (!connector_path) {
+		DLEYNA_LOG_DEBUG ("DLEYNA_CONNECTOR_PATH not set");
+		connector_path = CONNECTOR_DIR;
+	} else {
+		DLEYNA_LOG_DEBUG ("DLEYNA_CONNECTOR_PATH set to %s", connector_path);
+	}
+
+	connector_path_list = g_strsplit (connector_path, G_SEARCHPATH_SEPARATOR_S, 0);
+
+	for (i = 0; connector_path_list[i]; i++) {
+		path = g_strdup_printf("%s/%s%s.so", connector_path_list[i],
```

Comment:
For maximum compatibility you could probably use `g_module_build_path` here but that's minor and would also probably need a change for DLEYNA_CONNECTOR_LIB_PATTERN